pkg/internal/controller/controller.go

```diff
@@ -72,8 +70,8 @@ type Controller struct {
 	// the Queue for processing
 	Queue workqueue.RateLimitingInterface
 
-	// SetFields is used to inject dependencies into other objects such as Sources, EventHandlers and Predicates
-	SetFields func(i interface{}) error
+	// Dependencies are used to inject further dependencies into other objects, such as Sources, EventHandlers, and Predicates.
+	Dependencies inject.Context `inject:""`
```

Comment:
So this is the field users set in their main.go to provide external deps at the entry point.

Reply:
Yes, makes sense.